### PR TITLE
Allow searching for tournaments by name and abbreviation

### LINKED ISSUE #263: Allow search of tournaments by prefix
Users will want to be able to search for tournaments by the tournament prefix. Allow this by adding this functionality to the tournament search query (`where tournament.Abbreviation ILIKE {text}`)

### PR DESCRIPTION
**Pull request by foxtail286**
Closes #263, hopefully. 🤞

### CODE REVIEW

**Comment by foxtail286** [comment]
whoopsies, forgot to title the pull request :o

File: API/Repositories/Implementations/TournamentsRepository.cs [hunk 1]
```diff
@@ -23,7 +23,14 @@ public async Task<IEnumerable<Tournament>> SearchAsync(string name)
...
         name = name.Replace("_", @"\_");
-        return await _context.Tournaments.Where(x => EF.Functions.ILike(x.Name, $"%{name}%", @"\")).ToListAsync();
+
+        List<Tournament> nameSearchResult =
+        await _context.Tournaments.Where(x => EF.Functions.ILike(x.Name, $"%{name}%", @"\")).ToListAsync();
+
+        List<Tournament> abbreviationSearchResult =
+        await _context.Tournaments.Where(x => EF.Functions.ILike(x.Abbreviation, $"%{name}%", @"\")).ToListAsync();
+
+        return nameSearchResult.Union(abbreviationSearchResult);
```
Comment (hburn7) [inline API/Repositories/Implementations/TournamentsRepository.cs:33]: This is making two round trips when we don't need to. Let's combine these into something like this:
```cs
return await _context.Tournaments.Where(x => *ILike for name* || *ILike for abbreviation*);```

**Review by hburn7 — CHANGES_REQUESTED** [review]

**Comment by foxtail286** [comment]
Thanks for being patient, at least.

**Comment by hburn7** [comment]
Needs testing before merge, @myssto if you can.

**Comment by myssto** [comment]
will test

**Comment by hburn7** [comment]
Tested locally, seems fine.

**Pull request merged**